org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+            var eNewsData = qry.Select( item => item ).ToList();
+            HashSet<string> eNewsEmails = new HashSet<string>();
+            //string eNewsEmailList = "";
+            foreach ( var row in eNewsData )
+            {
+                string colVal = row.GetPropertyValue( "Email" ).ToStringSafe().ToLower();
+                if ( colVal != "" )
+                {
+                    eNewsEmails.Add( colVal );
+                    //eNewsEmailList = eNewsEmailList + "," + colVal;
+                }
+            }
+            Debug.WriteLine( string.Format("{0} Emails on ENews List.",eNewsData.Count ) );
+
+
+            HashSet<string> hsPersonEmails = new HashSet<string>();
+            foreach ( var contact in contacts )
+            {
+                if ( contact.properties.email.ToStringSafe() != null )
+                {
+                    hsPersonEmails.Add( contact.properties.email.ToStringSafe().ToLower() );
+                }
+            }
+
+            //WriteToLog( string.Format( "Total Contacts: {0}", contacts.Count() ) );
+            for ( var i = 0; i < contacts.Count(); i++ )
+            {
+                //Stopwatch watch = new Stopwatch();
+                //watch.Start();
+                Person person = personService.Get(contacts[i].properties.rock_person_id);
+
+                //For Testing
+                //WriteToLog( string.Format( "{1}i: {0}{1}", i, Environment.NewLine ) );
+                //WriteToLog( string.Format( "    After SQL: {0}{1}", watch.ElapsedMilliseconds, Environment.NewLine ) );
+
+                // If person is null, that means that we have a person in HS w/ a personId that no longer exists in rock
+                // This implies that a merge occurred and we need to look at the alias table to figure out what the new Id is and update it
+                // After updating the ID, we need to find the person object and handle patching
+
+                // Setup for patching
+                // Look up hubspot defined type
+                DefinedTypeService definedTypeService = new DefinedTypeService( _context );
+                DefinedType hsDefinedType = definedTypeService.Get( 527 ); // as of 7/8/24 Dev is 528, Train is 527
+                AttributeValueService attributeValueService = new AttributeValueService( _context );
+                AttributeService attributeService = new AttributeService( _context );
+
+
+                //Schedule HubSpot update if 1:1 match
+                if ( person != null )
+                {
+                    current_id = person.Id;
+                    var url = $"https://api.hubapi.com/crm/v3/objects/contacts/{contacts[i].id}";
+                    Debug.WriteLine( "Contact Count: " + i + " of " + contacts.Count() );
+                    Debug.WriteLine( "URL: " + url );
+                    var properties = new List<HubspotPropertyUpdate>();
+
+
+                    foreach ( DefinedValue hsSyncDv in hsDefinedType.DefinedValues )
```

Comment:
This is another place where we've got an abbreviation that is making the code harder to read. Please update the variable names to remove the abbreviations (e.g. `hsSyncDv` -> `hubSpotSyncDefinedValue`). The extra characters don't cost anything, and they improve readability.